cupy/random/cupy_distributions.cu

```diff
@@ -273,19 +273,23 @@ struct pointer_fnct {};
 
 template<typename T>
 struct pointer_fnct<true, T> {
-    using RetType = typename std::remove_pointer<T>::type;
-    static __device__ RetType get(T value, int id) {
-        return value[id];
+    static __device__ double get(int64_t *value, int id) {
```

Comment:
This trick is nice!
I was just planning on going all the way with Carray + Cindexer

Reply:
I think that using `CArray` is difficult because it has `int ndim` template parameter.

Reply:
Yes, I did a try by setting ndim always to 1 and making it contiguous, similarly to how the arrays are set to 1d before calling the elementwise function, but I thought it will be too much boilerplate code.